x-pack/plugin/snapshot-based-recoveries/src/test/java/org/elasticsearch/xpack/snapshotbasedrecoveries/recovery/plan/SnapshotsRecoveryPlannerServiceTests.java

```diff
@@ -217,18 +211,8 @@ public void testLogicallyEquivalentSnapshotIsUsedEvenIfFilesAreDifferent() throw
             final IndexVersion snapshotVersion;
             final Version luceneVersion;
             if (compatibleVersion) {
-                snapshotVersion = randomBoolean() ? null : IndexVersionUtils.randomCompatibleVersion(random());
-                // If snapshotVersion is not present,
-                // then lucene version must be < RecoverySettings.SEQ_NO_SNAPSHOT_RECOVERIES_SUPPORTED_VERSION
-                if (snapshotVersion == null) {
-                    luceneVersion = randomVersionBetween(
-                        random(),
-                        IndexVersions.V_7_0_0,
-                        RecoverySettings.SNAPSHOT_RECOVERIES_SUPPORTED_INDEX_VERSION
-                    ).luceneVersion();
-                } else {
-                    luceneVersion = IndexVersionUtils.randomCompatibleVersion(random()).luceneVersion();
-                }
+                snapshotVersion = IndexVersionUtils.randomCompatibleVersion(random());
+                luceneVersion = IndexVersionUtils.randomCompatibleVersion(random()).luceneVersion();
```

Comment:
I think this should be
```suggestion
                snapshotVersion = IndexVersionUtils.randomCompatibleVersion(random());
                luceneVersion = snapshotVersion.luceneVersion();
```
for version consistency? Otherwise the two versions can end up being unrelated. It may not fail the test (not sure). But seems strange to me.

Reply:
Yes. that makes sense to make the test more reliable.